src/pages/publication/contributors_tab/components/SortableTable.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC } from 'react';
+import { SortableContainer, SortableElement } from 'react-sortable-hoc';
+
+import {
+  Checkbox,
+  FormControlLabel,
+  Table,
+  TableBody,
+  TableCell,
+  TableRow,
+  TextField,
+  Button,
+} from '@material-ui/core';
+import { Contributor } from '../../../../types/contributor.types';
+import { FormikProps, useFormikContext, Field } from 'formik';
+import { Publication } from '../../../../types/publication.types';
+import AddContributor from '../AddContributor';
+import { useTranslation } from 'react-i18next';
+import DeleteIcon from '@material-ui/icons/Delete';
+
+import FormCardSubHeading from '../../../../components/FormCard/FormCardSubHeading';
+import { removeDuplicatesByScn, arrayMove } from '../../../../utils/helpers';
+
+interface SortableItemProps {
+  contributor: Contributor;
+  placement: number;
+  onDelete: (index: number) => void;
+}
+
+const SortableItem = SortableElement(({ contributor, placement, onDelete }: SortableItemProps) => {
+  const { t } = useTranslation();
+
+  const index = placement - 1;
+
+  return (
+    <TableRow tabIndex={0} key={contributor.systemControlNumber}>
+      <TableCell align="left">
+        <FormCardSubHeading>{contributor.name}</FormCardSubHeading>
+        <Field name={`contributors[${index}].corresponding`}>
+          {({ field }: any) => (
+            <FormControlLabel
+              control={<Checkbox checked={field.value} {...field} />}
+              label={t('publication:contributors.corresponding')}
+            />
+          )}
+        </Field>
+        <div>
+          {contributor.corresponding && (
+            <Field name={`contributors[${index}].email`}>
+              {({ field }: any) => <TextField variant="outlined" label={t('profile:email')} {...field} />}
+            </Field>
+          )}
+        </div>
+      </TableCell>
+      <TableCell align="left">
+        {contributor.institutions?.map(institution => (
+          <div key={`${institution.id}`}>{institution.name}</div>
+        ))}
+      </TableCell>
+      <TableCell align="right">
+        <div>
+          <div>{placement}</div>
+          <div>
+            <Button color="secondary" onClick={() => onDelete(index)}>
+              <DeleteIcon />
+              {t('common:remove')}
+            </Button>
+          </div>
+        </div>
+      </TableCell>
+    </TableRow>
+  );
+});
+
+interface SortableListProps {
+  contributors: Contributor[];
+  onDelete: (index: number) => void;
+  onSortEnd: ({ oldIndex, newIndex }: any) => void;
+}
+
+const SortableList = SortableContainer(({ contributors, onDelete }: SortableListProps) => {
+  const uniqueContributors = removeDuplicatesByScn(contributors);
```

Comment:
skal samme scn kunne inngå i flere bidragsytere, eller er dette bare en ekstra sikring?

Reply:
dette er forfattere i skjermbildet, og ikke bidragsytere - men du toucher innpå et viktig poeng. kanskje vi skal skille mellom forfattere og bidragsytere? 

vi må uansett filtrere ut at den samme forfatteren/bidragsyteren ikke kan være listet to ganger med den samme rollen, men som sagt - del 1 av x

Reply:
Spørsmålet er om en person kan ha flere roller. Hvis det skal være mulig må vi kunne registrere samme scn flere ganger, hvis ikke kan vi blokere for duplikater.